security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisImplProvider.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.contingency.ContingenciesProvider;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.security.interceptors.SecurityAnalysisInterceptor;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+@AutoService(SecurityAnalysisProvider.class)
+public class SecurityAnalysisImplProvider implements SecurityAnalysisProvider {
```

Comment:
Also, is it OK to leave it in the security-analysis-api module ?

The consequence is the following: you will always have that implementation in the classpath, which means that if you add another implementation you want to use, you will not be able to rely on the behaviour which allows to configure nothing to use the only implementation in the classpath.

Reply:
> `LoadFlowBasedSecurityAnalysis`

`DefaultSecurityAnalysis` would be fine. `LoadFlowBasedSecurityAnalysis` is maybe not appropriate as even an optimized security analysis is also based on a load flow.

Reply:
> Also, is it OK to leave it in the security-analysis-api module ?
> 
> The consequence is the following: you will always have that implementation in the classpath, which means that if you add another implementation you want to use, you will not be able to rely on the behaviour which allows to configure nothing to use the only implementation in the classpath.

You are right, it would break the new api design, as it would require to always have the default impl in the config.yml or use .find("h2")

Reply:
ManyLoadflowSecurityAnalysis ?

Reply:
Issue #1604 was created to keep these comments